docker.go

```diff
@@ -1603,3 +1604,20 @@ func (p *DockerProvider) SaveImages(ctx context.Context, output string, images .
 func (p *DockerProvider) PullImage(ctx context.Context, image string) error {
 	return p.attemptToPullImage(ctx, image, types.ImagePullOptions{})
 }
+
+var permanentClientErrors = []func(error) bool{
+	errdefs.IsNotFound,
+	errdefs.IsInvalidParameter,
+	errdefs.IsUnauthorized,
+	errdefs.IsForbidden,
+	errdefs.IsNotImplemented,
```

Comment:
Are all of the values covered by the tests? I can see errors for the NotFound and the InvalidParameter err, but I'm not able to find the other ones yet.

Reply:
No, I didn't check for all these conditions. I used one likely example condition for each retried client call but didn't want to repeat every condition in all calls since I think it's a bit noisy repeating all conditions in the tests, without much benefit.

I could check the `isPermanentClientError` function itself against all conditions but the function is also very simplistic and didn't seem to benefit much from checking the "completeness" of the list.

It's not an issue for me adding more testing. I have a few suggested approaches, what would you like to see:
* checking the `isPermanentClientError` itself for all expected conditions
* checking for other expectable conditions in the retry tests
* using another approach, being very specific in each client retry case on what errors can be expected and then test against that
* (another approach I did not think of?)

Reply:
As you shared, the helper method to check if the error is of the right type is very simplistic, so no need to add an unit test although it would be trivial.

For the rest of the errors not yet covered, I'd prefer having them as explicit tests. I know, we are creating a mock client which will produce those errors, but if at some point we reduce the list of the errors that are not permanent, I'd like to be warned by the tests.

Hope you understand

Reply:
Hey, I added more scenarios to include the other error types as well.